server/knowledge-graph/graph.service.js

```diff
@@ -63,10 +66,14 @@ class GraphService {
     const leafs = graph.getLeafNodes();
     // Trigger aggregations for leafs
     // Will be propagated to parent nodes
-    await Promise.map(profiles, profile => {
-      return Promise.map(leafs, node => profile.aggregateStats(node));
+    return Promise.each(chunk(profiles, CHUNK_SIZE), profilesChunk => {
+      return Promise.map(profilesChunk, async profile => {
+        await Promise.each(chunk(leafs, CHUNK_SIZE), leafsChunk => {
+          return Promise.map(leafsChunk, node => profile.aggregateStats(node));
+        });
+        return profile.save();
+      });
```

Comment:
Ideas here (and below):
- add concurrency with original map:
```js
Promise.map(profiles, profile => {}, { concurrency: CHUNK_SIZE })
```
- use promise queue to tackle `profile.save()` one by one

I would prefer concurrency, and I would personally set it to 5 since thats the maximum number of connections in a single pool (use case here) from the sequelize connection manager (postgre connection manager extends abstract):
https://github.com/sequelize/sequelize/blob/master/src/dialects/abstract/connection-manager.js#L32
This should ensure us we are never doing more than 5 "save" promises at a given point in time.

Would love to hear @underscope thoughts on this.